library/acl.inc

```diff
@@ -828,7 +828,9 @@
             return false;
         elseif($count_user_allowed > 0)
             return true;
-        elseif($count_group_denied > 0)
+        // We do not exclude user that belongs to group A and B
+        // In case of A group have permissions and B group does not have permissions
+        elseif($count_group_denied > 0 && $count_group_allowed < 1)
```

Comment:
This breaks a fundamental access rule. If there is a deny, then will not pass no matter what. Note that Openemr does not have any phpgacl DENY rules in it's standard setup; doing this requires setting it in Advanced phpgacl setup. And note that DENY is not the same as not having permission. So, if I am in 2 groups, and only 1 gives me permission, then it will work. Recommend removing this change since it should not be needed for your example.